tests/integration/shared/sources/new-or-changed-sources-test.js

```diff
@@ -0,0 +1,195 @@
+const path = require('path')
+const fs = require('fs')
+const test = require('tape')
+
+const srcShared = path.join(process.cwd(), 'src', 'shared')
+const datetime = require(path.join(srcShared, 'datetime', 'index.js'))
+const sourceMap = require(path.join(srcShared, 'sources', '_lib', 'source-map.js'))
+const changedSources = require(path.join(__dirname, '_lib', 'changed-sources.js'))
+
+const srcEvents = path.join(process.cwd(), 'src', 'events')
+const crawlerHandler = require(path.join(srcEvents, 'crawler', 'index.js')).handler
+const scraperHandler = require(path.join(srcEvents, 'scraper', 'index.js')).handler
+
+/** Test new or changed scrapers.
+ *
+ * By default, this checks new/changed sources by diffing the current branch
+ * with a baseline branch (see _lib/changed-sources.js).
+ *
+ * You can override this by setting some environment vars:
+ *
+ * TEST_ALL - tests all of the sources
+ * e.g., TEST_ALL=1 npm run test:integration
+ *
+ * TEST_ONLY - a comma-delimited list of sources to test
+ * TEST_ONLY=gb-sct,nl,gb-eng npm run test:integration
+ */
+
+process.env.NODE_ENV = 'testing'
+
+// A fake cache, destroyed and re-created for the test run.
+process.env.LI_CACHE_PATH = path.join(process.cwd(), 'zz-testing-fake-cache')
+
+
+//////////////////////////////////////////////////////////////////////
+// Utilities
+
+/** Split an array into batches, e.g.:
+ * makeBatches([1,2,3,4,5], 3) = [[1,2,3], [4,5]]
+ */
+function makeBatches(arr, batchsize) {
+  const ret = []
+  for (let i = 0; i < arr.length; i += batchsize)
+    ret.push(arr.slice(i, i + batchsize))
+  return ret
+}
+
+/** Runs the full crawl-scrape-save cycle for a given source,
+ * returning a struct indicating which steps succeeded. */
+async function runCrawlAndScrape(key, today) {
+  const result = {
+    key: key,
+    crawled: false,
+    scraped: false,
+    data: null,
+    written: false,
+    success: false,
+    error: null
+  }
+
+  try {
+    const crawlArg = {
+      Records: [
+        { Sns: { Message: JSON.stringify({source: key}) } }
+      ]
+    }
+    console.log(`Calling scrape for ${key}`)
+    await crawlerHandler(crawlArg)
+    result.crawled = true
+
+    const scrapeArg = {
+      Records: [
+        { Sns: { Message: JSON.stringify({source: key, date: today, silent: true}) } }
+      ]
+    }
+    const data = await scraperHandler(scrapeArg)
+    result.scraped = true
+    result.data = data
+
+    // TODO (testing) verify that data was actually written.
```

Comment:
Prob sufficient to just look at the data returned, as that is presumably what will be written to the DB.

Reply:
Right, yes.  Testing that it's written isn't valid here, as we're using mocked write destinations anyway.  Removing that, thanks.